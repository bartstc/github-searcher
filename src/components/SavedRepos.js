import React from 'react';
import { connect } from 'react-redux';
import { removeRepo } from '../modules/savedRepos/savedActions';
import SavedItem from './SavedItem';

export const SavedRepos = ({ repos, removeRepo }) => (
  <>
    {repos.length === 0
      ? <p>No saved repos</p>
      : (
        <>
          <h4 className="text-warning mt-5">Saved repos</h4>
          <ul className="list-group">
            {repos.map(repo => (
              <SavedItem {...repo} removeRepo={removeRepo} />
            ))}
          </ul>
        </>
      )
    }
  </>
);

export default connect(({ savedRepos }) => ({ repos: savedRepos }), { removeRepo })(SavedRepos);