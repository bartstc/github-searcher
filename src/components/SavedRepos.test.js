import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SavedRepos } from './SavedRepos';

describe('SavedRepos', () => {
  let props = {
    repos: [{
      owner: { login: 'test' },
      html_url: 'test',
      full_name: 'test',
      description: 'test',
      language: 'test'
    }]
  };
  let repos = shallow(<SavedRepos {...props} />);
  const ul = repos.find('.list-group');

  it('renders correctly', () => {
    expect(toJson(repos)).toMatchSnapshot();
  });

  it('renders ul with at least one SavedItem component', () => {
    expect(ul.find('SavedItem').exists()).toBe(true);
  });

  describe('when saved repos array is empty', () => {
    props = { repos: [] };
    repos = shallow(<SavedRepos {...props} />);

    it('return warning with text `No saved repos`', () => {
      expect(repos.find('p').text()).toEqual('No saved repos');
    });
  })
});