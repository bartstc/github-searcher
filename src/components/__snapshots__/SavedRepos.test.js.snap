// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedRepos renders correctly 1`] = `
<Fragment>
  <p>
    No saved repos
  </p>
</Fragment>
`;
